app/models/cd_api/defendant.rb

```diff
@@ -2,5 +2,9 @@
 
 module CdApi
   class Defendant < BaseModel
+    def linked?
+      maat_references = offence_summaries.map { |offence| offence&.laa_application&.reference }
+      maat_references.compact.first.present?
```

Comment:
I don't know how computationally intensive it is to find `offence&.laa_application&.reference` but if there is a significant cost (e.g. if there are lots of `offence_summaries` and `laa_applcation` has to perform a look-up) you can use a `lazy` enumerable. For example,

```suggestion
      maat_references = offence_summaries.lazy.map { |offence| offence&.laa_application&.reference }
      maat_references.any(&:present?)
```

This will stop looking for references when it has found one.